doc/examples/example2.md

```diff
@@ -0,0 +1,106 @@
+The distinguished name of the new certificate
+The root certificate details (_rootCertDetails) returns a unique pointer of DistinguishedName type. 
+
+\code{.cpp}
+_rootCertDetails = std::make_unique<DistinguishedName>(DistinguishedName::Builder{}
+        .commonName("ImATeapot")
+        .countryName("DE")
+        .organizationName("Linux AG")
+        .organizationalUnitName("Linux Support")
+        .pkcs9EmailAddress("support@example.com")
+        .localityName("oben")
+        .stateOrProvinceName("nebenan")
+        .serialNumber("08E36DD501941432358AFE8256BC6EFD")
+        .build());
+\endcode
+
+3- Serial Number
+The serial number of the new certificate
+
+4- Generate the Certificate Sign Parameters 
+The signing parameters for signing the new certificate
+Returns a CertificateSigningParameters type. 
+\code{.cpp}
+_caSignParams = CertificateSigningParameters::Builder{}
+        .certificateValidity(Asn1Time::Seconds(120))
+        .digestType(openssl::DigestTypes::SHA256)
+        .addExtension(caConstraint)
+        .addExtension(*_exampleUsage)
+        .build();
+\endcode
+
+4- Generate a Root RSA Certificate 
+
+With these parameters defined:
+- root RSA key (*_rootRSAKey)
+- the root certificate details (*_rootCertDetails)
+- the certificate authority parameters (_caSignParams)
+
+Becomes possible to generate a Root Certificate.
+
+The root RSA Certificate (_rootRsaCert) returns a unique pointer of X509Certificate type.  
+On this example zero (0) is passed as the serial number. 
+
+\code{.cpp}
+_rootRsaCert = std::make_unique<X509Certificate>(CertificateAuthority::createRootCertificate(
+                                                  *_rootRSAKey,
+                                                  *_rootCertDetails,
+                                                  0,
+                                                  _caSignParams));
+\endcode
+
+Certification Authority itself  
+------------------------------
+This code is going to generate additional information that is used when a certificate is signed.
+
+\code{.cpp}
+    _signParams = CertificateSigningParameters::Builder{}
+            .certificateValidity(Asn1Time::Seconds(120))
+            .notBeforeAsn1(Asn1Time::now())
+            .digestType(openssl::DigestTypes::SHA256)
+            .addExtension(*_exampleConstraints)
+            .addExtension(*_exampleUsage)
+            .build();
+\endcode
+
+The following code is an example on how creating a RSA certification authority.
+
+Parameters needed:
+
+- The default parameters for signing certificates ( _signParams)
+- The serial number which this CA should its first generated certificate (in this example is used 1)
+- Root Certificate the certificate of the CA (*_rootRsaCert)
+- Private Key the certificate's private key (*_rootRSAKey) 
+
+The return type is a certificate authority unique pointer(_rsaCa). 
+\code{.cpp}
+_rsaCa = std::make_unique<CertificateAuthority>(_signParams, 1, *_rootRsaCert, *_rootRSAKey);
```

Comment:
Add data type, remove _ from variable name. And no need to have a unique_ptr here. An honestly we might want to join this with the examples above just to make it easier to understand. Maybe we can have a big comprehensible example with comments, where we generate the keys, generate the self signed certificate, create the CA and sign some intermediate/client certificate.

Reply:
Agree.